src/Compilers/Core/Portable/Serialization/ObjectReader.cs

```diff
@@ -100,10 +101,20 @@ internal sealed partial class ObjectReader : IDisposable
                 return null;
             }
 
-            if (stream.ReadByte() != VersionByte1 ||
-                stream.ReadByte() != VersionByte2)
+            try
             {
-                return null;
+                if (stream.ReadByte() != VersionByte1 ||
+                    stream.ReadByte() != VersionByte2)
+                {
+                    return null;
+                }
+            }
+            catch (AggregateException ex)
+            {
+                // PipeReaderStream wraps any exception it throws in an AggregateException, which is not expected by
+                // callers treating it as a normal stream. Unwrap and rethrow the inner exception for clarity.
+                // https://github.com/dotnet/runtime/issues/70206
+                ExceptionDispatchInfo.Capture(ex.InnerException).Throw();
```

Comment:
Is it always safe to assume that ```InnerException``` is not null?

Reply:
➡️ Perhaps, but just in case I added a `when` clause

Reply:
`ExceptionDispatchInfo.Throw(ex.InnerException);` is a little simpler.

Reply:
➡️ Updated to use `ExceptionDispatchInfo.Throw` where available